scaleway/helpers_rdb.go

```diff
@@ -71,3 +72,101 @@ func expandInstanceSettings(i interface{}) []*rdb.InstanceSetting {
 
 	return res
 }
+
+func expandPrivateNetwork(data interface{}, exist bool) []*rdb.EndpointSpec {
+	if data == nil || !exist {
+		return nil
+	}
+
+	var res []*rdb.EndpointSpec
+	for _, pn := range data.([]interface{}) {
+		r := pn.(map[string]interface{})
+		spec := &rdb.EndpointSpec{
+			PrivateNetwork: &rdb.EndpointSpecPrivateNetwork{
+				PrivateNetworkID: expandID(r["pn_id"].(string)),
+				ServiceIP:        expandIPNet(r["ip"].(string)),
+			},
+		}
+		res = append(res, spec)
+	}
+
+	return res
+}
+
+func expandLoadBalancer() []*rdb.EndpointSpec {
+	var res []*rdb.EndpointSpec
+
+	res = append(res, &rdb.EndpointSpec{
+		LoadBalancer: &rdb.EndpointSpecLoadBalancer{}})
+
+	return res
+}
+
+func getRdbEndpointsToRemove(endPoints []*rdb.Endpoint, updates interface{}) (map[string]bool, error) {
+	actions := make(map[string]bool)
+	configs := make(map[string]*rdb.EndpointPrivateNetworkDetails)
+	privateIDToEndpointID := make(map[string]string)
+	for _, e := range endPoints {
+		// skip load balancer
+		if e.PrivateNetwork != nil {
+			pnZonedID := newZonedIDString(e.PrivateNetwork.Zone, e.PrivateNetwork.PrivateNetworkID)
+			actions[e.ID] = true
+			configs[pnZonedID] = e.PrivateNetwork
+			privateIDToEndpointID[pnZonedID] = e.ID
+		}
+	}
+
+	// compare private networks are persisted
+	for _, endpoint := range updates.([]interface{}) {
+		r := endpoint.(map[string]interface{})
+		locality, id, err := parseLocalizedID(r["pn_id"].(string))
+		if err != nil {
+			return nil, err
+		}
+		pn := &rdb.EndpointPrivateNetworkDetails{
+			PrivateNetworkID: id,
+			ServiceIP:        expandIPNet(r["ip"].(string)),
+			Zone:             scw.Zone(locality),
+		}
+
+		pnZonedID := newZonedIDString(pn.Zone, pn.PrivateNetworkID)
+		if _, configExist := configs[pnZonedID]; configExist {
+			isEqual := isEndPointEqual(configs[pnZonedID], pn)
+			// match the endpoint id for a private network
+			if _, endPointExist := privateIDToEndpointID[pnZonedID]; endPointExist {
+				actions[privateIDToEndpointID[pnZonedID]] = !isEqual
```

Comment:
A bit difficult to read and understand. Could you make it simpler to follow? We are at 3 imbrications levels. Could you give a more explicit name to configs for instance, is it a local config or a remote configuration?